extensions/src/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java

```diff
@@ -71,7 +71,17 @@ public UserListWrapper getUsers() {
   }
 
   public RoleListWrapper getRoles() {
-    return new RoleListWrapper( getUserRoleListService().getAllRoles() );
+    return getRoles( true );
+  }
+
+  public RoleListWrapper getRoles( boolean includeExtraRoles ) {
+    List<String> roles = getUserRoleListService().getAllRoles();
+    if ( !includeExtraRoles ) {
+      for ( String role : getExtraRoles() ) {
+        roles.remove( role );
```

Comment:
You are correct. This is a scenario where are removing extraRole from the list. If the user has a role name matching the extra role that we understand, it will be removed. I am open to suggestions here.